app/common/state/shieldState.js

```diff
@@ -0,0 +1,20 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const frameStateUtil = require('../../../js/state/frameStateUtil')
+const urlParse = require('../urlParse')
+
+function braveShieldsEnabled (frame) {
+  const activeRequestedLocation = frameStateUtil.activeRequestedLocation(frame)
```

Comment:
as a side node, `committed` does not mean that the url has completely loaded, it just means that a load was started and could have been blocked or redirected. Before `load-start` a url would be considered `provisional`